modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1554,44 +1546,66 @@ public int partitionId(BinaryRowEx row) {
     }
 
     /**
-     * Gathers the result of batch processing into a single resulting collection of rows.
+     * Gets a batch result.
      *
      * @param rowBatches Row batches.
      * @return Future of collecting results.
      */
-    static CompletableFuture<Collection<BinaryRow>> collectMultiRowsResponsesWithoutRestoreOrder(Collection<RowBatch> rowBatches) {
-        return allResultFutures(rowBatches)
-                .thenApply(response -> {
-                    var result = new ArrayList<BinaryRow>(rowBatches.size());
+    public static CompletableFuture<List<BinaryRow>> collectRejectedRowsResponsesWithRestoreOrder(Collection<RowBatch> rowBatches) {
+        return collectMultiRowsResponsesWithRestoreOrder(
+                rowBatches,
+                batch -> {
+                    List<BinaryRow> result = new ArrayList<>();
+                    List<BinaryRow> response = (List<BinaryRow>) batch.getCompletedResult();
```

Comment:
Why did you remove the previous version with the `Collection<Boolean>`?

Reply:
Because in the newest wersion we are using different approach, based on NullBinaryRow (for that row which applied to replication group and null for other).
Look at NullBinaryRow class to undestand it.

Reply:
Same about `BitSet`

Reply:
This would be implemented in another issue.

Reply:
Here it is
https://issues.apache.org/jira/browse/IGNITE-20669